pkg/koordlet/metriccache/metric_cache.go

```diff
@@ -74,11 +75,11 @@ type MetricCache interface {
 	Run(stopCh <-chan struct{}) error
 	TSDBStorage
 
-	GetNodeResourceMetric(param *QueryParam) NodeResourceQueryResult
 	GetPodResourceMetric(podUID *string, param *QueryParam) PodResourceQueryResult
 	GetContainerResourceMetric(containerID *string, param *QueryParam) ContainerResourceQueryResult
 	GetNodeCPUInfo(param *QueryParam) (*NodeCPUInfo, error)
 	GetNodeLocalStorageInfo(param *QueryParam) (*NodeLocalStorageInfo, error)
+	GetNodeDeviceInfo() (*NodeDeviceInfo, error)
```

Comment:
for this struct type, like(NodeDeviceInfo, NodeCPUInfo, NodeLocalStorageInfor),
we can provide a KVStorage interface (e.g. implemented by map[string]interface{}) with Get(obj) and Set(obj) functions.

```go
type MetricCache interface {
  TSDBStorage
  KVStorage
}
```